### DISCUSSION TITLE
Dynamically linked vcpkg in statically linked application

### DISCUSSION THREAD

**Discussion by az-faro** [microsoft/vcpkg#38433] [Q&A]
I have an application built in Visual Studio using `/MTd`. I also use vcpkg with `Use Static Libraries - No`. As far as I understand this should work fine, right? The application is statically linked with my own packages, but it uses vcpkg packages as dlls.

In either case I have not had issues using either the nlohmann package or miniz with this configuration.

However I recently added poco and now in the (debug) linking phase I get `LINK : fatal error LNK1104: cannot open file 'PocoFoundationmtd.lib'`. Now why would it want to open the `mtd` variant? shouldn't it be using the `d` variant? Am I doing something wrong or is the Poco package bugged?

**Comment by dg0yt** [comment]
Previous discussion: #38462
